bears/vcs/mercurial/HgCommitBear.py

```diff
@@ -0,0 +1,60 @@
+import re
+import shutil
+import os
+
+from urllib.parse import urlparse
+from coala_utils.ContextManagers import change_directory
+from coalib.misc.Shell import run_shell_command
+from bears.vcs.CommitBear import _CommitBear
+
+
+class HgCommitBear(_CommitBear):
+    LANGUAGES = {'Hg'}
+    SUPPORTED_HOST_KEYWORD_REGEX = {
+        'bitbucket': (r'[Cc]los(?:e[sd]?|ing)'
+                      r'|[Rr]esolv(?:e[sd]?|ing)'
+                      r'|[Ff]ix(?:e[sd]|ing)?')
+    }
+
+    @classmethod
+    def check_prerequisites(cls):
+        if shutil.which('hg') is None:
+            return 'hg is not installed.'
+        else:
+            return True
+
+    @staticmethod
+    def get_host_from_remotes():
+        """
+        Retrieve the first host from the list of mercurial remotes.
+        """
+        remotes, _ = run_shell_command('hg paths')
+
+        remotes = [url.split()[-1] for url in remotes.splitlines()]
+        if len(remotes) == 0:
+            return None
+
+        url = remotes[0]
+        if 'hg@' in url:
+            netloc = re.findall(r'@(\S+)\.', url)[0]
+        else:
+            netloc = urlparse(url)[1]
+        return netloc.split('.')[0].split('@')[-1]
+
+    def get_head_commit(self):
+        stdout = None
+        stderr = None
+        with change_directory(self.get_config_dir() or os.getcwd()):
+            stdout, stderr = run_shell_command(
+                'hg log -l 1 --template "{desc}"')
+        return (stdout, stderr)
+
+    def get_issue_regex(self, body_close_issue_full_url, host):
+        if body_close_issue_full_url:
+            result_info = 'full issue'
+            issue_ref_regex = (
+                r'https?://{}\S+/issues/(\S+)/\S+'.format(re.escape(host)))
```

Comment:
(reviewing this as it relates to https://github.com/coala/coala-bears/pull/2459 )

The trailing `/\S+` is optional, and I expect some repos would prefer to prevent it because they make the line very long.

e.g. https://bitbucket.org/vinay.sajip/sarge/issues/38/python-37-breakage vs https://bitbucket.org/vinay.sajip/sarge/issues/38 or https://bitbucket.org/vinay.sajip/sarge/issues/38/ , and could even be complete gibberish like https://bitbucket.org/vinay.sajip/sarge/issues/38/this-isnt-checked-by-bitbucket

Note that github allows https://github.com/coala/coala-bears/issues/2396/ , but not https://github.com/coala/coala-bears/issues/2396/foobar , and gitlab is the same with https://gitlab.com/coala/mobans/issues/6/ and not https://gitlab.com/coala/mobans/issues/6/foobar

Reply:
Also this is not related to Hg at all.

https://bitbucket.org/jayvdb/coala-projects/issues/1/testing-1-2-3 is a git repository hosted on BitBucket, and it also has the title after the issue number.

Reply:
Also according to https://github.com/coala/coala-bears/issues/1134 / https://confluence.atlassian.com/bitbucket/resolve-issues-automatically-when-users-push-code-221451126.html , the full URL will not close the issue , so 'full issue' is not supported by bitbucket.

Reply:
>and could even be complete gibberish like https://bitbucket.org/vinay.sajip/sarge/issues/38/this-isnt-checked-by-bitbucket

Oh okay
so preferably its sensible to simple drop the extra '/\S+' at the end of this regex if I understood it correctly ?

And I am not aware of any other hg hoster, so full issue url is a complete mess for hg bear since bitbucket doesn't support that, so should we dump this setting alltogether ?

Reply:
yup; full issue shouldnt be supported by HgCommitBear until we find a hoster that supports closing using them.